frontends/verific/verific.cc

```diff
@@ -450,8 +450,30 @@ void VerificImporter::import_attributes(dict<RTLIL::IdString, RTLIL::Const> &att
 		auto type_range = nl->GetTypeRange(obj->Name());
 		if (!type_range)
 			return;
+		if (type_range->IsTypeScalar()) {
+		  const long long left_bound = type_range->GetScalarRangeLeftBound();
+		  const long long right_bound = type_range->GetScalarRangeRightBound();
+		  const unsigned bit_width = type_range->NumElements();
+		  
+		  const std::bitset<sizeof(long long) * CHAR_BIT> binary_left(left_bound);
+		  log_assert(bit_width <= binary_left.size());
+		  std::vector<bool> bits_left(bit_width);
+		  for (size_t i = 0; i < bit_width; ++i) {
+		    bits_left[i] = binary_left[i];
+		  }
+		  
+		  const std::bitset<sizeof(long long) * CHAR_BIT> binary_right(right_bound);
+		  log_assert(bit_width <= binary_right.size());
+		  std::vector<bool> bits_right(bit_width);
+		  for (size_t i = 0; i < bit_width; ++i) {
+		    bits_right[i] = binary_right[i];
+		  }
+		  
+		  attributes.emplace(ID(left_bound), RTLIL::Const(bits_left));
```

Comment:
How do we tell from RTLIL if the value should be interpreted signed or unsigned?

Reply:
Good point. Actually, in a first iteration, it was automatically detected because all the bounds were 32 bit integers.
Now that they can have arbitrary width, we have to specify signedness.
Last commit adds a `is_bound_signed` just for that.

Reply:
We could use the signed flag on Const

Reply:
The issue with using the signed flag on Const is that info gets lost if we `write_rtlil` and then `read_rtlil`.
If we read the attributes in the same yosys session there is no issue, but I think we should also support exporting to RTLIL

Reply:
> The issue with using the signed flag on Const is that info gets lost if we `write_rtlil` and then `read_rtlil`.

I opened an issue #4539